### ISSUE TITLE
Crash when using nested classes for generalization

### ISSUE THREAD

**Issue by TimLethbridge** [umple/umple#1159]
Umple allows one class to be nested inside another as a way to represent generalization. We believe this feature is not commonly used, but it is useful to illustrate large generalization hierarchies, or for cases where one wants to add a subclass with just a minor difference from a superclass.

However there is a bug in parsing in the following case.

```
class TopHat{
  class Question{
   boolean status;
  }

 class HomeworkQuestion{
    isA Question;
    Date dueDate;
  }
}
```

The following is the stack trace:

```
Exception java.lang.NullPointerException in
java.lang.NullPointerException
	at cruise.umple.compiler.UmpleInternalParser.checkExtendsForCycles(UmpleInternalParser.java:2527)
	at cruise.umple.compiler.UmpleInternalParser.postTokenClassAnalysis(UmpleInternalParser.java:2021)
	at cruise.umple.compiler.UmpleInternalParser.postTokenAnalysis(UmpleInternalParser.java:1286)
	at cruise.umple.compiler.UmpleInternalParser.analyze(UmpleInternalParser.java:1067)
	at cruise.umple.compiler.UmpleModel.run(UmpleModel.java:1463)
	at cruise.umple.UmpleConsoleMain.runConsole(UmpleConsoleMain.java:143)
	at cruise.umple.UmpleConsoleMain.main(UmpleConsoleMain.java:204)

```

**Comment by TimLethbridge** [comment]
Here is another case. I am going to mark this as easy, as I think this should help make this debuggable.

```
class Hospital {
class Hospital {

}
}
```

**Comment by TimLethbridge** [comment]
Fixed by #1201
